app/vmselect/promql/exec.go

```diff
@@ -452,6 +453,11 @@ func isSubQueryComplete(e metricsql.Expr, isSubExpr bool) bool {
 			return false
 		}
 		for _, arg := range exp.Args {
+			if re, ok := arg.(*metricsql.RollupExpr); ok {
+				if re.Window != nil {
```

Comment:
Should we also check for wrong expr like `sum(up[:5m])`?

Reply:
Good point! Please see updates in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6450/commits/296902003ac1cd8b5220f6832d3f9b52705b05c9

Note, it now forbids use of `rate(metric)` when no ranged vector is specified - it is what you suggested from the very beginning. I think you were right to restrict this.